src/Adapter/Order/OrderAmountUpdater.php

```diff
@@ -129,9 +140,14 @@ public function update(
             // Update order details (if quantity or product price have been modified)
             $this->updateOrderDetails($order, $cart);
 
+            $productsComparator = new CartProductsComparator($cart);
```

Comment:
Could you add it to constructor please ?

Reply:
I don't think I can, I need to call this with the `$cart` state as is it at that time.

Reply:
Not a good idea because it depends on the Cart, and this one is a method parameter :)